### PR TITLE
Migration

### PR DESCRIPTION
**Pull request by iamhectorsosa**
This PR introduces:
* Replacement of app Dir and custom docs implementation in favor of Nextra
* Replacement of all local icons in favor of Nextra and Lucide icons
* Removal of Preview Mode in favor of in-page preview (future plans for Preview mode are still in motion)

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        destination: "/templates/minimal/overview",
+        permanent: false,
+      },
+      {
+        source: "/templates/minimal",
+        destination: "/templates/minimal/overview",
+        permanent: false,
+      },
+    ];
+  },
```
Comment (ericvalcik) [inline next.config.js:31]: We have a lot of redirects here, I think we could remove a significant part of them if we could move the `Overview` pages from `/docs/guide/overview` to just `/docs/guide` as we tried at one point. So the final structure would look like this:

image

I know we had some issues with this, and I don't know how much time you spend on making this work so if you feel like getting this to work is not worth it let me know 😆.

**Review by ericvalcik — APPROVED** [review]
I love the changes, the design is cleaner in my opinion too. About the comments I made, I'd create issues from some of them, but I felt like we can discuss them in this PR before we make issues from the comments I made.

I think the best would be to just discuss some of the possible changes here, but let's not make the changes in this PR - let's create GitHub issues for the stuff we will want to fix and fix those issues in new, smaller PRs.

👍 Approved, nice work.

**Review by ericvalcik — COMMENTED** [review]
Let's not merge yet tho, until we address all the comments and what to do next.

File: .vercelignore [hunk 2]
```diff
@@ -1 +0,0 @@
-packages/cli
```
[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply .vercelignore:1]: Correct

File: src/components/landing/Homepage.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+          method="post"
+        >
+          <input
+            className="w-full rounded-xl p-4 text-base lg:w-96"
+            type="email"
+            name="email_address"
+            placeholder="Your email"
+            aria-label="email"
+          />
+          <CTA>Subscribe</CTA>
```
[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply src/components/landing/Homepage.tsx:78]: Yeah, using Zod here would be nice to add some form validation. https://github.com/webscopeio/mailingui/issues/214

[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply src/components/landing/Homepage.tsx:78]: Yeah, using Zod here would be nice to add some form validation. https://github.com/webscopeio/mailingui/issues/214

File: src/components/shared/CodeBlock.tsx
[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply src/components/shared/CodeBlock.tsx:1]: Actually, a dude in Nextra shared with me a project where we imported Nextra's Pre and Code components. So we might end up actually using everything the same. Opening an issue to give it a shot https://github.com/webscopeio/mailingui/issues/215

File: src/components/shared/CodeBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable tailwindcss/no-custom-classname */
```
[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply src/components/shared/CodeBlock.tsx:1]: We still do want to catch these in other files. We aren't disabling this that much around.

File: tsconfig.json [hunk 5]
```diff
@@ -9,7 +9,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
```
[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply tsconfig.json:12]: I'd also like to know. Brought from migration project.

File: next.config.js (hunk 1 above)
Comment (iamhectorsosa) [reply next.config.js:31]: Let's give it a shot https://github.com/webscopeio/mailingui/issues/216

File: theme.config.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+    const ogImage = asPath.includes("/blog")
+      ? "images/og/blog.png"
+      : asPath.includes("/docs")
+      ? "images/og/components.png"
+      : asPath.includes("/feedback")
+      ? "images/og/feedback.png"
+      : asPath.includes("/templates")
+      ? "images/og/templates.png"
+      : "images/og/homepage.png";
```
[the remark this answers is not in the document]
Comment (iamhectorsosa) [reply theme.config.jsx]: 100%

**Pull request merged**